delta/extensions/metrics.py

```diff
@@ -30,7 +30,7 @@ class SparseMetric(tensorflow.keras.metrics.Metric): # pylint:disable=abstract-m
     An abstract class for metrics applied to integer class labels,
     with networks that output one-hot encoding.
     """
-    def __init__(self, label, class_id: int=None, name: str=None, binary: int=False):
+    def __init__(self, label=None, class_id: int=None, name: str=None, binary: int=False, label_id: int=None, **kwargs):
```

Comment:
the class_id method should be able to take either a string or an integer, so probably this is unnecessary?

Reply:
Do you mean to just remove the int after class_id?

Reply:
No, I meant remove the label_id argument entirely, because the label argument can already be a string or an int

Reply:
Should I replace all references to self._label_id with self._class_id ?

Reply:
If I make those changes then the model fails to load.  Maybe cleaning up label_id should be saved for another PR?

Reply:
Sorry if my last comment wasn't clear, I changed my mind and think it is ok as is. Can you just add label_id to the function documentation and mention it should only be used internally, and then I think we are good to go.